allennlp/common/util.py

```diff
@@ -57,8 +57,9 @@ def pad_sequence_to_length(sequence: List,
             padded_sequence.insert(0, default_value())
     return padded_sequence
 
+A = TypeVar('A')
 
-def add_noise_to_dict_values(dictionary: Dict[Any, float], noise_param: float) -> Dict[Any, float]:
+def add_noise_to_dict_values(dictionary: Dict[A, float], noise_param: float) -> Dict[A, float]:
```

Comment:
Why use `TypeVar` instead of `Any`?  To ensure that the return type matches the input type?  (My "why" questions in this review aren't generally complaints, it's just wanting to understand mypy's type checking better.)